tests/phpunit/Unit/Components/NavbarHorizontal/PersonalToolsTest.php

```diff
@@ -79,4 +79,38 @@ public function testGetHtml_LoggedInUserHasNoNewMessages( $domElement ) {
 		$this->assertNotTag( $matcher, $instance->getHtml() );
 	}
 
+	/**
+	 * @covers ::getHtml
+	 * @dataProvider domElementProviderFromSyntheticLayoutFiles
+	 */
+	public function testGetHtml_LoggedOutUserHasNewMessages( $domElement ) {
+		$factory = MockupFactory::makeFactory( $this );
+		$factory->set( 'UserIsLoggedIn', false );
+		$chameleonTemplate = $factory->getChameleonSkinTemplateStub();
+		$chameleonTemplate->data = [ 'newtalk' => 'foo' ];
+
+		/** @var Component $instance */
+		$instance = new $this->classUnderTest( $chameleonTemplate, $domElement );
+
+		$matcher = [ 'class' => 'pt-anontalk' ];
+		$this->assertTag( $matcher, $instance->getHtml() );
```

Comment:
I think this code is actually clearer if you do not have a dedicated variable for matcher.

Reply:
I followed the pattern used by other unit tests. Should I rather follow your suggestion for all new tests (including this PR), but leave old tests as they are (with $matcher)?